### ISSUE TITLE
SC2006 doesn't catch deferred backticks in PS1

### ISSUE THREAD

**Issue by ssokolow** [koalaman/shellcheck#836]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2006
- My shellcheck version (`shellcheck --version` or "online"): 0.3.2
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

PS1="`hostname`"
PS1='`hostname`'
```

(Though, in practice, it'd likely be something more along the lines of a "get current git branch" function)

#### Here's what shellcheck currently says:
```
Line 3:
PS1="`hostname`"
     ^-- SC2006: Use $(..) instead of legacy `..`.
```

#### Here's what I wanted or expected to see:
```
Line 3:
PS1="`hostname`"
     ^-- SC2006: Use $(..) instead of legacy `..`.

Line 4:
PS1='`hostname`'
     ^-- SC2006: Use $(..) instead of legacy `..`.
```

Since checks like SC2025 already recognize the need for special behaviour in prompts, SC2006 should be aware of the deferred evaluation behaviour of `PS1` and complain about backticks in single quotes too.
